### ISSUE TITLE
Reviews loaded through API

### ISSUE THREAD

**Issue by jabberdog** [jabberdog/Wled#1]
I think this is a good moment to start investing in preparing a solution so that we can load reviews through their API because of 2 reasons:

1. We anyway want to have a bit of control over the length of the reviews loaded as on some pages that looks ugly in case the reviews are longer. So if manage to get them loaded via API maybe you can control and specify in case a review is longer than say 150 characters have a short button saying More or any of that.

2. Lately we have been some issues with their scripts which seemed like not loading properly the reviews and back and forth and so maybe this way we get them straightforward and also improve the speed because whenever they have issues the product page loads horribly because of their script.

We will need to integrate a method on the site that allows users to create reviews for these products directly from the product page.

**Comment by jabberdog** [comment]
To return a list of reviews for the product, it is possible to send a GET request to this endpoint: 
https://api.reviews.co.uk/product/review?store=wholesale-led-lights&sku={SKU}

{SKU} should be replaced by the SKU of the product that you wish to get the review for (e.g. 
MS-GU103W)

The response body is returned in JSON and simply requires converting either into an associative array or XML object in order to sort through the data and return which parts we want. 

I am assuming that we want to collate the same information that is shown on the page right now:

 - Average (out of 5)
 - Stars (out of 5)
 - Total reviews for product
 - Name of reviewer
 - Review

I have attached a file which contains the data that is returned once we have made a request to their API. If there is anything else in that data that you want to add to the list above then please comment.

**Comment by jabberdog** [comment]
Liam, if possible for you what I would also add is the info whether the reviewer is or not a verified buyer so that we can display this information for these verified buyers that reviewed products and so build up trust.

Thanks,
Florin

**Comment by jabberdog** [comment]
Also add to that list of fields you pull out the DATE when the reviews were made as I assume you would need to sort them by that in the end.

**Comment by jabberdog** [comment]
Documentation: https://jabberdog.github.io/ReviewsApi/

**Comment by jabberdog** [comment]
I spoke with Amit this morning and we need you to style the blocks for the reviews so that we can load the reviews through API and get rid of the script provided by reviews.co.uk (I know we left this task at the point where you managed to pull them out through APIs, we established what data we needed to play around with, but we did not finish styling the block with the reviews so that we can load them).

So, the question to you is can you work tonight on styling that? You can go by a similar styling like the one we now have from reviews.co.uk.

This issue becomes critical as this morning we noticed that because of the reviews script not loading properly (they had some technical issues) our ajax cart was not loading and so the conversion went massively down.

Also regarding the procedure to pull out the reviews, please optimize the procedure and ensure that when it happens that our requests to reviews.co.uk are not processed successfully, none of the scripts and elements within our pages will not be impacted by that request not being responded accordingly.

Thanks,
Florin

**Comment by jabberdog** [comment]
Ok, so I've added the reviews API to Wholesale LED Lights. I've quickly made it look similar to the original reviews widget that we were using.

There is an issue with the amount of reviews returned by them via their API which I need to talk to them about (they are only returning 15 max right now). But I think this is fine for now.

If we really want to show an average rating based on ALL the reviews for the product then I'll have to do the maths on that, so it isn't done yet on version 1.0.

If you have any issues let me know. Amit is also aware.

**Comment by jabberdog** [comment]
We need to have them all so yes that one needs to be set out with Reviews.co.uk as soon as we can because it will hurt a bit on best selling products which have 100+ reviews both because of customers less trust, but also because search engines won't see nothing pointing to a place where they presumably could be loaded like it was the case before.

Apart from that we really need to show the average rating & total count as explicit as possible. We can't afford to let neither the customer try and figure that out by doing the maths themselves as they are used to get an overall figure (this is clear expectation they have for products), neither the search engines like G as G expects in the reviews area to see an aggregated figure which should present the total count, average overall rating figure then match those figures with the actual no. of reviews into the page. And the 2 aggregated figures need to be made with big font size, standing out enough so that can be seen as important both by clients & search engines into the page.

As discussed when started this, we decided to truncate the reviews so that they wound't create a long list there but we need to have a little button at the end of each that should allow the customer to see the full review/expand the remaining part of it.

Once all the reviews will be pulled out from reviews not just the 15 as is now, we need to have pagination like we did with their script and have them all available to our customers to browse and read.

**Comment by jabberdog** [comment]
Liam for time being until we have the full solution on reviews I think is safer for us to stay on the old Reviews script and so I commented the code you worked last night on view.phtml on the test-product template and removed the comment you put on the reviews script area so that we can use that for time being.

I think apart from the issues listed in my first comment below (just to be sure you are aware of and will take that into account too) I need to emphasise the fact that in the final solution we will use we need to avoid to store all reviews into the product page as was in the solution built so far and have them loaded on request if technically possible for you (that one can come up when the user clicks pagination to see next reviews etc) as otherwise just thinking we are at risk to mess it up with that many blocks of texts (user generated content into the product page which can harm the overall quality of the page).

So basically what I'm trying to say is that we need to avoid having them simply as the output of the php that now runs there. Lets say we can opt in having maximum 5 reviews displayed by default then the rest loaded on request.

**Comment by jabberdog** [comment]
One last thing to also have in mind is that we need to allow them add a review like we do now through the script reviews.co.uk provides, so you will also have to implement the review form from the reviews.co.uk so that they will still be able to add a review starting from our product page which gets saved on reviews.co.uk database like it does now.
